frontend/components/player/playing_controls.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { GrCirclePlay } from "react-icons/gr";
+import { BiSkipPrevious, BiSkipNext, BiShuffle } from "react-icons/bi";
+import { BsRepeat, BsRepeat1, BsPauseCircle } from "react-icons/bs";
+import { useEffect } from "react";
+
+const PlayingControls = ({
+	isPlaying,
+	trackProgress,
+	togglePlay,
+	toPrevTrack,
+	toNextTrack,
+	toggleShuffle,
+}) => {
+	useEffect(() => {}, [isPlaying]);
+	return (
+		<div className="playing-controls">
+			<BiShuffle
+				className="player__grey-icon repeat-shuffle-icon"
+				aria-label="Shuffle"
+				onClick={toggleShuffle}
+			/>
+			<BiSkipPrevious
+				className="player__grey-icon"
+				aria-label="Previous"
+				onClick={toPrevTrack}
+			/>
+			{isPlaying ? (
+				<BsPauseCircle
+					className="player__white-icon"
+					aria-label="Pause"
+					onClick={togglePlay}
```

Comment:
Since both the `pause` button and the `play` button take a `togglePlay`. There may be a case in the future that you might want to stop playing for some other reason. Might be worth considering refactoring this to something along the lines of `setDesiredPlayState(true)`, or just explicitly a `play()` & `pause()`.

Reply:
Noting your feedback. Currently this is a Redux action intended solely as an onClick for the user to press play or pause. The HTMLAudioElement does have built-in `play()` and `pause()` functions that I am actually currently using, just abstracted away from the user.